dictionary.py

```diff
@@ -75,24 +83,88 @@
         name = dicts["first_name"]
         if dicts["salary_rub"] > 100000:
             print(name)
-            
+
 # 5 Вывести позиции, на которых люди получают меньше 80к (можно с повторениями).
 for dictionary in departments:
     for dicts in dictionary["employers"]:
         position = dicts["position"]
         if dicts["salary_rub"] < 80000:
-            print(position) 
+            print(position)
 
-# 6 Посчитать, сколько денег в месяц уходит на каждый отдел – и вывести вместе с названием отдела 
+# 6 Посчитать, сколько денег в месяц уходит на каждый отдел – и вывести вместе с названием отдела
 
 
 for dictionary in departments:
     dep_spendings = list()
     for dicts in dictionary["employers"]:
         sallary = dicts["salary_rub"]
         dep_spendings.append(sallary)
-                    
+
         final_dep_spendings = sum(dep_spendings)
     print(f"затраты {dictionary['title']} - {final_dep_spendings}")
 
-    
+# 7. Вывести названия отделов с указанием минимальной зарплаты в нём.
+for employers_info in departments:
+    all_sallary = list()
+    for info in employers_info["employers"]:
+        sallary = info["salary_rub"]
+        all_sallary.append(sallary)
+        minimum_sallry = min(all_sallary)
+    print(
+        f"минимальная зарплата в отделе {employers_info['title']} : {minimum_sallry}")
+
+
+# 8. Вывести названия отделов с указанием минимальной, средней и максимальной зарплаты в нём.
+ultimate_sallary = list()
+for employers_info in departments:
+    all_sallary = list()
+    for info in employers_info["employers"]:
+        sallary = info["salary_rub"]
+        all_sallary.append(sallary)
+        minimum_sallry = min(all_sallary)
+        midium_sallry = sum(all_sallary)/len(all_sallary)
+        maximum_sallary = max(all_sallary)
+    print(f"""минимальная зарплата в отделе {employers_info['title']}:{minimum_sallry},
+           средняя: {midium_sallry}, максимальная: {maximum_sallary}""")
+
+# 9. Вывести среднюю зарплату по всей компании.
+    ultimate_sallary.extend(all_sallary)
+    medium_ultimate_sallary = max(ultimate_sallary)/len(ultimate_sallary)
+print(medium_ultimate_sallary)
+# 10. Вывести названия должностей, которые получают больше 90к без повторений.
+positioin_set = set()
+for employers_info in departments:
+    for info in employers_info["employers"]:
+        position = info["position"]
+        if info["salary_rub"] > 90000:
+            positioin_set.add(position)
+
+for position_ in positioin_set:
+    print(position_)
+
+
+# 11. Посчитать среднюю зарплату по каждому отделу среди девушек
+# (их зовут Мишель, Николь, Кристина и Кейтлин).
+for employers_info in departments:
+    all_women_sallary = list()
+    for info in employers_info["employers"]:
+        sallary = info["salary_rub"]
+        name = info["first_name"]
+        if (name == "Michelle" or name == "Nicole" or name == "Caitlin"
```

Comment:
Это работает, но выглядит сложно и трудно читается. Для проверки вхождения предпочтительно использовать структуры, предназначенные для этого. Например множества.
```suggestion
        women_names = { "Michelle", "Nicole", "Caitlin", "Christina"}
        if name in women_names:
```